packages/block-editor/src/components/off-canvas-editor/block-select-button.js

```diff
@@ -62,14 +62,23 @@ function ListViewBlockSelectButton(
 		}
 	}
 
-	const editAriaLabel = blockInformation
+	let editAriaLabel = blockInformation
 		? sprintf(
 				// translators: %s: The title of the block.
 				__( 'Edit %s block' ),
 				blockInformation.title
 		  )
 		: __( 'Edit' );
 
+	editAriaLabel =
+		block.attributes?.id && block.attributes?.label
+			? sprintf(
+					// translators: %s: The title of the page.
+					__( 'View %s' ),
+					block.attributes?.label
+			  )
+			: editAriaLabel;
+
```

Comment:
Can I check this has been tested in the Nav block list view as well that appears in the Inspector controls when selecting the Navigation block?

This exhibits different behaviour - namely when you click an item it opens the block settings for that block. Therefore does "View" make sense in that context?

Reply:
> Can I check this has been tested in the Nav block list view as well that appears in the Inspector controls when selecting the Navigation block?

I am not sure if I understood it correctly, but @getdave can you please check below screenshot. Is that what you wanted to check? Apologies if I misunderstood it.
<img width="1439" alt="Screenshot 2023-03-07 at 7 38 50 PM" src="https://user-images.githubusercontent.com/40795917/223446763-5fceccb6-acb9-4784-a9f8-7a2a986338ac.png">

Reply:
Ah no I mean [this thing](https://www.youtube.com/watch?v=VAxn8AhFcLs&t=110s)

Reply:
Thanks @getdave 
In the list view as well, it is showing the same updates in the title.